csrc/host_ir/jit.cpp

```diff
@@ -0,0 +1,383 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <bfs.h>
+#include <fusion.h>
+#include <global_allocator.h>
+#include <host_ir/executor.h>
+#include <ir/all_nodes.h>
+#include <ops/all_ops.h>
+#include <val_graph_visitor.h>
+
+#include <instrumentation.h>
+#include <llvm/ExecutionEngine/JITLink/JITLink.h>
+#include <llvm/ExecutionEngine/Orc/CompileUtils.h>
+#include <llvm/ExecutionEngine/Orc/IRCompileLayer.h>
+#include <llvm/ExecutionEngine/Orc/LLJIT.h>
+#include <llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h>
+#include <llvm/ExecutionEngine/Orc/ThreadSafeModule.h>
+#include <chrono>
+#include <queue>
+#include <unordered_map>
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <ATen/ATen.h>
+#include <c10/core/MemoryFormat.h> // for c10::optional
+#include <host_ir/jit.h>
+#include <functional>
+#include <memory>
+
+// Include the actual header files instead of forward declarations
+#include <multidevice/communicator.h>
+#include <runtime/fusion_executor_cache.h>
+#include <runtime/fusion_kernel_runtime.h>
+
+namespace nvfuser {
+
+using allocate_fn =
+    at::Tensor* (*)(const int64_t*, int64_t, const int64_t*, int64_t);
+
+class HostIrJitParams {
+  private:
+  Communicator* communicator_;
+  hir::HostIrEvaluatorParams params_;
+  // Cache Fusions, KernelExecutors
+  std::unordered_map<hir::HostUnit*, std::unique_ptr<ExecutorAbstract>> executors_;
+  std::unordered_map<hir::HostUnit*, FusionExecutorCache> fec_;
+  using StreamKey = std::variant<int64_t, hir::Stream*>;
+  std::unordered_map<StreamKey, c10::cuda::CUDAStream> streams_;
+  std::unordered_map<Expr*, c10::intrusive_ptr<c10d::Work>> works_;
+  const int64_t my_local_device_index_;
+  // IpcHandleCache ipc_handle_cache_;
```

Comment:
I might be missing something here. But I don't see [meaningful] uses of these variables in this PR.